xds/src/test/java/io/grpc/xds/LrsLoadBalancerTest.java

```diff
@@ -0,0 +1,325 @@
+    List<EquivalentAddressGroup> backendAddrs = createResolvedBackendAddresses(2);
+    deliverResolvedAddresses(backendAddrs, "round_robin");
+    FakeLoadBalancer childBalancer = (FakeLoadBalancer) childBalancers.poll();
+    assertThat(childBalancer.name).isEqualTo("round_robin");
+    deliverResolvedAddresses(backendAddrs, "pick_first");
+    assertThat(childBalancer.shutdown).isTrue();
+    childBalancer = (FakeLoadBalancer) childBalancers.poll();
+    assertThat(childBalancer.name).isEqualTo("pick_first");
+    loadBalancer.shutdown();
+    assertThat(childBalancer.shutdown).isTrue();
+  }
+
+  @Test
+  public void errorPropagation() {
+    loadBalancer.handleNameResolutionError(Status.UNKNOWN.withDescription("I failed"));
+    ArgumentCaptor<SubchannelPicker> pickerCaptor = ArgumentCaptor.forClass(null);
+    verify(helper)
+        .updateBalancingState(eq(ConnectivityState.TRANSIENT_FAILURE), pickerCaptor.capture());
+    Status status =
+        pickerCaptor.getValue().pickSubchannel(mock(PickSubchannelArgs.class)).getStatus();
+    assertThat(status.getDescription()).contains("I failed");
+
+    List<EquivalentAddressGroup> backendAddrs = createResolvedBackendAddresses(2);
+    deliverResolvedAddresses(backendAddrs, "round_robin");
+    loadBalancer.handleNameResolutionError(Status.UNKNOWN.withDescription("I failed"));
+    verify(helper, times(2))
+        .updateBalancingState(eq(ConnectivityState.TRANSIENT_FAILURE), pickerCaptor.capture());
+    status = pickerCaptor.getValue().pickSubchannel(mock(PickSubchannelArgs.class)).getStatus();
+    assertThat(status.getDescription()).contains("I failed");
+    assertThat(status.getDescription()).contains("handled by downstream balancer");
+  }
+
+  private void deliverResolvedAddresses(
+      List<EquivalentAddressGroup> addresses, String childPolicy) {
+    PolicySelection childPolicyConfig =
+        new PolicySelection(new FakeLoadBalancerProvider(childPolicy), null, null);
+    LrsConfig config =
+        new LrsConfig(
+            CLUSTER_NAME, EDS_SERVICE_NAME, LRS_SERVER_NAME, TEST_LOCALITY, childPolicyConfig);
+    ResolvedAddresses resolvedAddresses =
+        ResolvedAddresses.newBuilder()
+            .setAddresses(addresses)
+            .setAttributes(
+                Attributes.newBuilder()
+                    .set(XdsAttributes.ATTR_CLUSTER_SERVICE_LOAD_STATS_STORE, loadRecorder)
+                    .build())
+            .setLoadBalancingPolicyConfig(config)
+            .build();
+    loadBalancer.handleResolvedAddresses(resolvedAddresses);
+  }
+
+  private static List<EquivalentAddressGroup> createResolvedBackendAddresses(int n) {
+    List<EquivalentAddressGroup> list = new ArrayList<>();
+    for (int i = 0; i < n; i++) {
+      SocketAddress addr = new FakeSocketAddress("fake-address-" + i);
+      list.add(new EquivalentAddressGroup(addr));
+    }
+    return list;
+  }
+
+  private final class FakeLoadBalancerProvider extends LoadBalancerProvider {
```

Comment:
You can use `TestUtils.StandardLoadBalancerProvider` after pulling master.

Reply:
No, I do not like that.